src/librustc_mir/build/scope.rs

```diff
@@ -366,6 +383,15 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
         let len = self.scopes.len();
         assert!(scope_count < len, "should not use `exit_scope` to pop ALL scopes");
         let tmp = self.get_unit_temp();
+
+        // If we are emitting a `drop` statement, we need to have the cached
+        // diverge cleanup pads ready in case that drop panics.
+        let may_panic =
+            self.scopes[(len - scope_count)..].iter().any(|s| s.drops.iter().any(|s| s.kind.may_panic()));
```

Comment:
This line is >100 chars, failing `tidy`. (I consider a for-loop + break is easier to read here.)

```
[00:03:50] tidy error: /checkout/src/librustc_mir/build/scope.rs:390: line longer than 100 chars
[00:03:51] some tidy checks failed
```